sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobAsyncClient.java

```diff
@@ -0,0 +1,468 @@
+     * @throws NullPointerException thrown if the {@code data} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<UploadManifestResult> uploadManifest(BinaryData data) {
+        if (data == null) {
+            return monoError(logger, new NullPointerException("'data' can't be null."));
+        }
+
+        return withContext(context -> this.uploadManifestWithResponse(data.toFluxByteBuffer(), context)).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Uploads a manifest to the repository.
+     * The client currently only supports uploading OciManifests to the repository.
+     * And this operation makes the assumption that the data provided is a valid OCI manifest.
+     * <p>
+     * Also, the data is read into memory and then an upload operation is performed as a single operation.
+     * @see <a href="https://github.com/opencontainers/image-spec/blob/main/manifest.md">Oci Manifest Specification</a>
+     *
+     * @param data The manifest that needs to be uploaded.
+     * @return The rest response containing the operation result.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code data} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<UploadManifestResult>> uploadManifestWithResponse(BinaryData data) {
+        if (data == null) {
+            return monoError(logger, new NullPointerException("'data' can't be null."));
+        }
+
+        return withContext(context -> this.uploadManifestWithResponse(data.toFluxByteBuffer(), context));
+    }
+
+    Mono<Response<UploadManifestResult>> uploadManifestWithResponse(Flux<ByteBuffer> data, Context context) {
+        if (data == null) {
+            return monoError(logger, new NullPointerException("'data' can't be null."));
+        }
+
+        BufferedFlux playableFlux = new BufferedFlux();
+
+        return data.concatMap(playableFlux::write)
+            .then(Mono.defer(() -> Mono.just(playableFlux.getDigest())))
+            .flatMap(dig -> this.registriesImpl.createManifestWithResponseAsync(
+                repositoryName,
+                dig,
+                playableFlux.flush(),
+                playableFlux.getSize(),
+                UtilsImpl.OCI_MANIFEST_MEDIA_TYPE,
+                context))
+            .flatMap(response -> {
+                Response<UploadManifestResult> res = new ResponseBase<ContainerRegistriesCreateManifestHeaders, UploadManifestResult>(
+                    response.getRequest(),
+                    response.getStatusCode(),
+                    response.getHeaders(),
+                    new UploadManifestResult(response.getDeserializedHeaders().getDockerContentDigest()),
+                    response.getDeserializedHeaders());
+
+                return Mono.just(res);
+            }).onErrorMap(UtilsImpl::mapException);
```

Comment:
Why does this method take `Flux<ByteBuffer>` as input? The public APIs either take `BinaryData` or `OciManifest` which is then converted to `Flux<ByteBuffer>` which may not be required if we use `ByteBuffer` as the param for this method. Then BinaryData can be converted using `BinaryData.toByteBuffer()` and OciManifest can be converted using.

```java
ByteBuffer byteBuffer = ByteBuffer.wrap(this.registryImplClient.getSerializerAdapter().serializeToBytes(manifest, SerializerEncoding.JSON));
```
Suggesting this because the doc says that the content will be fully loaded into memory, so, BinaryData can be mapped to ByteBuffer instead of `Flux`.

Reply:
Yes, the current implementation is to load it into memory. I did the following for 2 reasons

(a) Future proofing.
(b) Autorest by default creates Rest API argument as Flux<ByteBuffer>, which I believe can be changed.

happy to revert if you think it would be better performance wise.  However I expect both of them to be similar - no?